homeassistant/components/switch/rachio.py

```diff
@@ -0,0 +1,225 @@
+import logging
+import voluptuous as vol
+from datetime import timedelta
+
+import homeassistant.helpers.config_validation as cv
+import homeassistant.util as util
+from homeassistant.components.switch import SwitchDevice, PLATFORM_SCHEMA
+from homeassistant.const import CONF_ACCESS_TOKEN
+
+REQUIREMENTS = ['https://github.com/Klikini/rachiopy'
+                '/archive/edf666de8ef3f9596ddc8d3a989e8c4b829a4319.zip'
+                '#rachiopy==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'switch'
+
+DATA_RACHIO = 'rachio'
+
+CONF_manual_run_mins = 'manual_run_mins'
+manual_run_mins = 60
+
+STATUS_ONLINE = 'ONLINE'
+
+MIN_UPDATE_INTERVAL = timedelta(minutes=5)
+MIN_FORCED_UPDATE_INTERVAL = timedelta(seconds=1)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ACCESS_TOKEN): cv.string
+})
+
+_IRO = None
+
+
+# Set up the component
+# noinspection PyUnusedLocal
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    global _IRO
```

Comment:
Don't use global, use `hass.data[DATA_RACHIO ]` instead, which is a dict for this purpose.